api/src/db-connection.ts

```diff
@@ -77,30 +78,61 @@ export class Connection {
     return await this._knex("logentry").insert(logEntry);
   }
 
-  async getAllPapers() {
-    const response = await this._knex.raw<{ rows: PaperWithEntityCounts[] }>(`
-      SELECT paper.*,
-             (
-                SELECT COUNT(*)
-                  FROM citation
-                 WHERE citation.paper_id = paper.s2_id
-             ) AS citations,
-             (
-                SELECT COUNT(*)
-                  FROM symbol
-                 WHERE symbol.paper_id = paper.s2_id
-             ) AS symbols
+  async getAllPapers(offset: number = 0, size: number = 25): Promise<Paginated<PaperIdWithEntityCounts>> {
+    // We use a local type to capture the fact that counts will come across the wire
+    // as a string.
+    type Row = PaperIdWithEntityCounts & {
+        symbol_count: string;
+        citation_count: string;
+        sentence_count: string;
+        term_count: string;
+        equation_count: string;
+        definition_count: string;
+        entity_count: string;
+        total_count: string;
+    };
+    const response = await this._knex.raw<{ rows: Row[] }>(`
+      SELECT paper.arxiv_id,
+             paper.s2_id,
+             version.index AS version,
+             SUM(CASE WHEN entity.type = 'symbol' THEN 1 ELSE 0 END) AS symbol_count,
+             SUM(CASE WHEN entity.type = 'citation' THEN 1 ELSE 0 END) AS citation_count,
+             SUM(CASE WHEN entity.type = 'sentence' THEN 1 ELSE 0 END) AS sentence_count,
+             SUM(CASE WHEN entity.type = 'term' THEN 1 ELSE 0 END) AS term_count,
+             SUM(CASE WHEN entity.type = 'equation' THEN 1 ELSE 0 END) AS equation_count,
+             SUM(CASE WHEN entity.type = 'definition' THEN 1 ELSE 0 END) AS definition_count,
```

Comment:
fyi: the way definitions are stored in the database is about to change with my next pipeline update. They will be sorted as rows in the `entitydata` table, where each row is with a 'term' or 'symbol' entity from the `entity` table.

Maybe it's worth removing this column for now? Or, if we want to include it anyway, we can put together a new query to get definition counts once I've incorporated the pipeline updates.

Reply:
Sure. I have some work on it's way that'll fit nicely with this. I'll remove the column in that PR.

Reply:
sounds good, thank you!